test/OrchardCore.Commerce.Tests.UI/Tests/InventoryTests/InventoryBehaviourTests.cs

```diff
@@ -73,6 +75,56 @@ public InventoryBehaviourTests(ITestOutputHelper testOutputHelper)
             },
             browser);
 
+    [Theory, Chrome]
+    public Task UnavailableProductInCartShouldPreventCheckout(Browser browser) =>
+        ExecuteTestAfterSetupAsync(
+            async context =>
+            {
+                await context.SignInDirectlyAndGoToRelativeUrlAsync("/testproduct");
+                await context.ClickReliablyOnAsync(By.XPath("//button[contains(., 'Add to cart')]"));
+
+                // Set Inventory of product in cart to 0.
+                await context.GoToAdminRelativeUrlAsync($"/Contents/ContentItems/{TestProduct}/Edit");
+                await context.ClickReliablyOnAsync(By.XPath("//label[contains(., 'Allows Back Order')]"));
+                await context.ClickAndFillInWithRetriesAsync(By.Name("InventoryPart.Inventory[TESTPRODUCT]"), "0");
+                await context.ClickPublishAsync();
+
+                // Verify checkout is unavailable.
+                await context.GoToCartAsync();
+                context.Exists(By.XPath($"//p[contains(., '{CheckoutUnavailableMessage}')]"));
+                await context.GoToCheckoutAsync();
+                context.Exists(By.XPath($"//div[contains(., '{CheckoutUnavailableMessage}')]"));
```

Comment:
Avoid just checking div text, because there are lots of divs on the page. Check for the class first like
`context.Exists(By.XPath($"//div[contains(@class, 'message-error') and contains(., '{CheckoutUnavailableMessage}')]"));`